### DISCUSSION TITLE
Issue with connecting locally to Fastify server in WSL2 Win 11 Pro

### DISCUSSION THREAD

**Discussion by smolinari** [fastify/fastify#3229] [Q&A] [answered]
Hi everyone,

I'm working on a project that is using Fastify with NestJS and can't get connection to the server. Any attempts to connect end in a retry connection error. 

I'm using WSL2 (Ubuntu 20.04 distro) on Win 11 Pro and Node 14.17.4. 

Things I've tried:

1. asked other devs to run this: https://github.com/smolinari/test-nestjs-merc 
 - result: 3 devs on Mac and Linux (native) all confirmed it works fine, so it is a problem with WSL2 and Windows.  
2. setting 0.0.0.0 in the startup for possibly binding the IP address differently (as suggested by the NestJS docs) - didn't work
2. Removing WSL2 and reinstalling it, including Ubuntu 20.04. Still didn't work.  
3. Running an express based NestJS app. It works fine/ as expected. 
4. Also installed OpenSuse to try it out. Still no go. 
5. Turning on Altair. No connectivity there either.

I'm sort of pulling my hair out as I don't have any real notion as to why this might be happening and thus don't know how to troubleshoot further. (and please don't say, just change to Mac or Linux LOL! 😁) 

Does anyone have any tips to try and troubleshoot this or may know of the issue off the bat and maybe even a solution?

Scott

**Comment by jsumners** [comment]
Have you tried any of the answers in https://stackoverflow.com/questions/61002681/connecting-to-wsl2-server-via-local-network

**Reply by smolinari** [reply]
Hey thanks for replying @jsumners. 

I found out the issue. It seems Windows was taking up port 3000. Why no errors were given, I don't know. Why I didn't just try a different port too, I don't know. But, lesson learned. Try a different port for SnGs just to be sure. 

Scott

**Comment by smolinari** [comment, marked as answer]
Ok. Seems it was a silly cause to my issue. A Windows service was already taking port 3000 (why that is, I still need to find out). At any rate, of course the silly resolution was to pick a different port and everything works now. 

Scott
